core/field.js

```diff
@@ -64,1155 +64,1143 @@ goog.require('Blockly.Gesture');
 
 /**
  * Abstract class for an editable field.
- * @param {*} value The initial value of the field.
- * @param {?Function=} opt_validator  A function that is called to validate
- *    changes to the field's value. Takes in a value & returns a validated
- *    value, or null to abort the change.
- * @param {Object=} opt_config A map of options used to configure the field. See
- *    the individual field's documentation for a list of properties this
- *    parameter supports.
- * @constructor
- * @abstract
- * @implements {IASTNodeLocationSvg}
- * @implements {IASTNodeLocationWithBlock}
- * @implements {IKeyboardAccessible}
- * @implements {IRegistrable}
- * @alias Blockly.Field
  */
-const Field = function(value, opt_validator, opt_config) {
+class Field {
   /**
-   * A generic value possessed by the field.
-   * Should generally be non-null, only null when the field is created.
-   * @type {*}
-   * @protected
+   * @abstract
+   * @implements {IASTNodeLocationSvg}
+   * @implements {IASTNodeLocationWithBlock}
+   * @implements {IKeyboardAccessible}
+   * @implements {IRegistrable}
+   * @alias Blockly.Field
    */
-  this.value_ = this.DEFAULT_VALUE;
+  constructor() {
+    /**
+     * Name of field.  Unique within each block.
+     * Static labels are usually unnamed.
+     * @type {string|undefined}
+     */
+    this.name = undefined;
```

Comment:
Yeah that would be sweet! But it would definitely have to be a bigger refactor because names are actually optional (most FieldLabels don't have names for example). And names are actually passed to the Input rather than the Field itself via the [appendField](https://github.com/google/blockly/blob/e21a8fcd0cf16fd26b589d6fbf42d469ca57e863/core/input.js#L103) function.

As a bit of trivia, Field names should always be unique within a block, because they are used for lookup, but lookup is done using a [fun nested loop](https://github.com/google/blockly/blob/e21a8fcd0cf16fd26b589d6fbf42d469ca57e863/core/block.js#L1071) rather than a dictionary.